magento/magento2
# Fix PHPDocs for \Magento\Framework\Data\Collection::getItemById

Add null as possible type returned by `getItemById` method.
Method returns null when item is not found.

PHPStan complains in the code using that method:
Sample usage:
```php
$product = $productCollection->getItemById(42);
if ($product) { # phpstan: If condition is always true.
    // some logic
}
```

Result in PHPStorm:
![image](https://github.com/magento/magento2/assets/6697660/76c2f16b-0bc3-44cc-9966-18626e0b284d)

### Description (*)

Adds `null` type to PHPDocs of `\Magento\Framework\Data\Collection::getItemById`.
Fixes minor grammar issues in PHPDocs of that class.
### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Create a module that uses any Collection and add code which reflects `if` condition specified in the PR summary
2. Configure PHPStan with enabled `Treat PHPDoc types as certain` (usually it is enabled by default)
3. Run static analysis (or open file in IDE with static analysis capabilities such as PHPStorm)
4. There should be no warning/error `if condition is always true`

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#38485: Fix PHPDocs for \Magento\Framework\Data\Collection::getItemById
